examples/echo/echo_requester.cpp

```diff
@@ -0,0 +1,250 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file implements a chip-echo-requester, for the
+ *      CHIP Echo Protocol.
+ *
+ *      The CHIP Echo Protocol implements two simple methods, in the
+ *      style of ICMP ECHO REQUEST and ECHO REPLY, in which a sent
+ *      payload is turned around by the responder and echoed back to
+ *      the originator.
+ *
+ */
+
+#include "common.h"
+
+#include <protocols/echo/Echo.h>
+#include <support/ErrorStr.h>
+
+#define ECHO_CLIENT_PORT (CHIP_PORT + 1)
+
+namespace {
+
+// Max value for the number of EchoRequests sent.
+constexpr size_t kMaxEchoCount = 3;
+
+// The CHIP Echo interval time in microseconds.
+constexpr int32_t gEchoInterval = 1000000;
+
+// The EchoClient object.
+Protocols::EchoClient gEchoClient;
+
+SecureSessionMgr<Transport::UDP> gSessionManager;
+
+Inet::IPAddress gDestAddr;
+
+// The last time a CHIP Echo was attempted to be sent.
+uint64_t gLastEchoTime = 0;
+
+// True, if the EchoClient is waiting for an EchoResponse
+// after sending an EchoRequest, false otherwise.
+bool gWaitingForEchoResp = false;
+
+// Count of the number of EchoRequests sent.
+uint64_t gEchoCount = 0;
+
+// Count of the number of EchoResponses received.
+uint64_t gEchoRespCount = 0;
+
+uint64_t Now(void)
+{
+    struct timeval now;
+    gettimeofday(&now, NULL);
+
+    return ((uint64_t) now.tv_sec * 1000000) + (uint64_t) now.tv_usec;
```

Comment:
C++-style casts, please, and in the rest of the code too.

Reply:
Acked